File: artemis-entity-tracker-gui/src/net/namekdev/entity_tracker/ui/model/ManagerTableModel.java
package net.namekdev.entity_tracker.ui.model;

import javax.swing.table.DefaultTableModel;

public class ManagerTableModel extends DefaultTableModel {
	public ManagerTableModel() {
		addColumn("");
		addColumn("manager");
	}

	@Override
	public Class<?> getColumnClass(int columnIndex) {
		switch (columnIndex) {
			case 0: return Boolean.class;
			case 1: return String.class;
			default: return null;
		}
	}

	public boolean isCellEditable(int row, int column) {
		return column == 0;
	}

	public void addManager(String name) {
		addRow(new Object[] { true, name });
	}

	public String getManagerName(int index) {
		return (String) getValueAt(index, 1);
	}

	public boolean getManagerState(int index) {
		return (boolean) getValueAt(index, 0);
	}
}

File: artemis-entity-tracker-gui/src/main/java/net/namekdev/entity_tracker/ui/model/ManagerTableModel.kt
package net.namekdev.entity_tracker.ui.model

class ManagerTableModel : BaseSystemTableModel() {
    init {
        columnIdentifiers[1] = "manager"
    }

    fun addManager(name: String) {
        addRow(arrayOf(true, name))
    }

    fun getManagerName(index: Int): String {
        return getValueAt(index, 1) as String
    }

    fun getManagerState(index: Int): Boolean {
        return getValueAt(index, 0) as Boolean
    }

    override fun clear() {
        rowCount = 0
    }
}